python/lsst/ap/association/association.py

```diff
@@ -75,27 +78,173 @@ def _set_flux_stats(dia_object_record, dia_sources, filter_name, filter_id):
+
+    noNanMask = np.logical_and(np.isfinite(fluxes), np.isfinite(fluxErrors))
+    fluxes = fluxes[noNanMask]
+    fluxErrors = fluxErrors[noNanMask]
+    midpointTais = dia_sources.get("midPointTai")[currentFluxMask][noNanMask]
+
+    if len(fluxes) == 1:
+        dia_object_record['%sPSFluxMean' % filter_name] = fluxes
+        dia_object_record['%sPSFluxNdata' % filter_name] = 1
+    elif len(fluxes > 1):
+        fluxMean = np.average(fluxes, weights=1 / fluxErrors ** 2)
+
+        # Standard, DDPD defined columns.
+        dia_object_record['%sPSFluxMean' % filter_name] = fluxMean
+        dia_object_record['%sPSFluxMeanErr' % filter_name] = np.sqrt(
+            1 / np.sum(1 / fluxErrors ** 2))
+        dia_object_record['%sPSFluxSigma' % filter_name] = np.std(fluxes,
+                                                                  ddof=1)
         dia_object_record["%sPSFluxChi2" % filter_name] = np.sum(
             ((fluxMean - fluxes) / fluxErrors) ** 2)
         dia_object_record['%sPSFluxNdata' % filter_name] = len(fluxes)
 
+        # Columns below are created in DM-18316 for use in ap_pipe/verify
+        # testing.
+        ptiles = np.percentile(fluxes, [5, 25, 50, 75, 95])
+        dia_object_record['%sPSFluxPercentile05' % filter_name] = ptiles[0]
+        dia_object_record['%sPSFluxPercentile25' % filter_name] = ptiles[1]
+        dia_object_record['%sPSFluxMedian' % filter_name] = ptiles[2]
+        dia_object_record['%sPSFluxPercentile75' % filter_name] = ptiles[3]
+        dia_object_record['%sPSFluxPercentile95' % filter_name] = ptiles[4]
+
+        dia_object_record['%sPSFluxMAD' % filter_name] = \
+            median_absolute_deviation(fluxes)
+
+        dia_object_record['%sPSFluxSkew' % filter_name] = skew(fluxes)
+
+        dia_object_record['%sPSFluxMin' % filter_name] = fluxes.min()
+        dia_object_record['%sPSFluxMax' % filter_name] = fluxes.max()
+
+        deltaFluxes = fluxes[1:] - fluxes[:-1]
+        deltaTimes = midpointTais[1:] - midpointTais[:-1]
+        dia_object_record['%sPSFluxMaxSlope' % filter_name] = np.max(
+            deltaFluxes / deltaTimes)
+
+        m, b = _fit_linear_flux_model(fluxes, fluxErrors, midpointTais)
+        dia_object_record['%sPSFluxLinearSlope' % filter_name] = m
+        dia_object_record['%sPSFluxLinearIntercept' % filter_name] = b
+
+        dia_object_record['%sPSFluxStetsonJ' % filter_name] = _stentson_J(
+            fluxes, fluxErrors)
+
+        dia_object_record['%sPSFluxErrMean' % filter_name] = \
+            np.mean(fluxErrors)
+
+    totFluxes = dia_sources.get("totFlux")[currentFluxMask]
+    totFluxErrors = dia_sources.get("totFluxErr")[currentFluxMask]
+    noNanMask = np.logical_and(np.isfinite(totFluxes),
+                               np.isfinite(totFluxErrors))
+    totFluxes = totFluxes[noNanMask]
+    totFluxErrors = totFluxErrors[noNanMask]
+
+    if len(totFluxes) == 1:
+        dia_object_record['%sFPFluxMean' % filter_name] = totFluxes
```

Comment:
Let's call these `totFluxMean` and so on so the link to the `DIASource` quantity is more clear

Reply:
That's what I initially had too, however, FPFlux is coming from the current version of the cat schema and the schema in dax_ppdb. I'm trying to reuse those two as much as possible so things don't get even more confused with the column names. I can add totFlux columns to the schema extra files and just leave the FPF names empty but that might be confusing for folks using the current setup. Hopefully once we have a automated way of making yaml schemas from the DPDD soon.

Reply:
I just tend to think FPFlux should refer to a different thing, which is means/errors on the diffim forced fluxes.

Reply:
Okay, so would you prefer that I add them in as specifically totFlux here and in the schema files?

Reply:
Yes, that would be my preference